src/types/createResolve.ts

```diff
@@ -7,7 +7,10 @@ export default function createResolve<TFields extends string, TContext = any>(
   fieldNames: TFields[],
 ): GraphQLFieldConfig<any, TContext>['resolve'] {
   return async (obj, args, context, info) => {
-    const nodeType = info.parentType as NodeType<any, Record<TFields, any>>;
+    const nodeType = info.parentType as NodeType<
+      any,
+      Record<TFields | 'id', any>
```

Comment:
I think this is probably fine. Like why not just include `id` -- even if it may not exist on the node its a fine tradeoff imo

Reply:
is there no way we can leverage `TSource` from the object class here? http://www.staging-typescript.org/play?target=7#code/C4TwDgpgBAyg9gVwE4GNoF4oG8oDM5wBcUAzsEgJYB2A5lAL4DcAUM6JLIqhAEoQlwANgDcISADwAVeMjQA+KJgAUA2RGLSuaAJSKFZSrRZtw0Gdz4CRYgIJIkAQxBTz8xVDuPnr3vyGiJTTU5OWNcBCoUYAo4KigUHhctCDklXGIVZI0fXXR9cmoaXSxmKDKoJAhgZDjcFnpWFFiyKAdiH0t-W3sncR8FTABtJRx8OAZchTGAGnieJRG8Agm9JbhtAF1GIA

Reply:
Wont something like that require every consumer of `createResolve` to have had defined an actual Source interface type?

Reply:
bump @taion ?

Reply:
i was hoping that `createResolve` could infer `TSource` based on the object type in which it's used, assuming we define `TSource` on that object type (and default it to `{ id: string }`). does that not work?

Reply:
See my latest push. It's kind of wonky